src/runtime_src/core/tools/xbutil2/SubCmdValidate.cpp

```diff
@@ -1003,7 +1004,13 @@ p2pTest(const std::shared_ptr<xrt_core::device>& _dev, boost::property_tree::ptr
 
   std::string msg;
   XBU::xclbin_lock xclbin_lock(_dev);
-  XBU::check_p2p_config(_dev.get(), msg);
+  std::vector<std::string> config;
+  try {
+    config = xrt_core::device_query<xrt_core::query::p2p_config>(_dev);
+  }
+  catch (const xrt_core::query::no_such_key&) {  }
```

Comment:
As best I can tell based on pipeline logs is that xbutil validate fails on u50 because it calls device_query   on line 1009, the key exists, but the sysfs node cannot be opened.  The latter results in std::runtime_error, which is not caught and causes xbutil to exit with the exception message:
`[xbutil] ERROR: Failed to open /sys/bus/pci/devices/0000:3b:00.1/p2p.u.11534336/config for reading: No such file or directory`

I am not liking that the sysfs getters throws generic std::runtime_error, I think we should have a specific exception for sysfs error, maybe` xrt_core::query::sysfs_error`, similar to `xrt_core::query::no_such_key`.  If we derive both off of a `xrt_core::query::error `exception then most code that catches `...`, or `std::runtime_error` can be changed to catch xrt_core::query::error instead.

Reply:
@AShivangi Please check #5531 for exception hierarchy for query errors.  If you catch xrt_core::query::exception& then you catch errors related to query requests (no_such_key, sysfs_error, a few more), but you will not catch unrelated "real" errors.